hypha/core/mail.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict, List, Optional
+
+from anymail.message import AnymailMessage
+from django.conf import settings
+from django.core.mail import get_connection
+from django.core.mail.backends.base import BaseEmailBackend
+from django.http import HttpRequest
+from django.template.loader import get_template
+from mdmail.api import EmailContent  # type: ignore
+
+
+def send_mail(
+    subject: str,
+    message: str,
+    from_email: str,
+    recipient_list: List[str],
+    subject_prefix: Optional[str] = settings.EMAIL_SUBJECT_PREFIX,
+    context: Optional[Dict] = None,
+    request: Optional[HttpRequest] = None,
+    fail_silently: bool = False,
+    css: Optional[str] = None,
+    image_root: str = ".",
+    auth_user: Optional[str] = None,
+    auth_password: Optional[str] = None,
+    connection: Optional[BaseEmailBackend] = None,
+    reply_to: Optional[List[str]] = None,
+) -> AnymailMessage:
+    """Drop in replacement for django.core.email.send_mail.
+
+    This replaces django's django.core.email.send_mail, but the mail will have
+    an html alternative rendered from the text part with markdown.
+
+    You can also provide a custom css and even images (that will be inlined)
+    located in image_root.
+
+    If you want to write your markdown in a template, just put the name of the
+    template in message and add a context (which can be {}) and eventually a request:
+    """
+    connection = connection or get_connection(
+        username=auth_user,
+        password=auth_password,
+        fail_silently=fail_silently,
+    )
+    if context is not None:
+        message = get_template(message).render(context, request)
+    if subject_prefix:
+        subject = f"{subject_prefix}{subject}"
+    content = EmailContent(message, css=css, image_root=image_root)
+    mail = AnymailMessage(
+        subject,
+        content.text,
+        from_email,
+        recipient_list,
+        connection=connection,
+        reply_to=reply_to,
+    )
+    mail.attach_alternative(content.html, "text/html")
+
+    mail.send()
+    return mail
```

Comment:
This seems to ignore the `EMAIL_BACKEND` setting. Is it really needed just to send out HTML e-mails? Can we not extend the existing `send_mail()` with `EmailContent` instead?

Reply:
`AnymailMessage` provides some standardized error codes per email provider, like mailgun, which is stored in the database, not here but when triggered via the celery task in the EmailAdapter. I tried to keep the interface the same. 

This function is essentially a re-write of the default `send_mail` provided by Django, which wrappers around the `EmailMessage` class to send text emails without HTML as attachment.

Reply:
Done, implement as `MarkdownMail()` class with a send() function